### PR TITLE
Documenting the Module Test Recipes into modules.md for module developers

### PR DESCRIPTION
**Pull request by MariusNi**
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: docs/modules.md [hunk 1]
```diff
@@ -971,37 +971,79 @@ The code for a module can be split into a static library and a shared object (SO
...
+- ModuleObject::Open with same signature as MmiOpen
+- ModuleObject::Close with same signature as MmiClose
+- ModuleObject::GetInfo with same signature as MmiGetInfo
 - ModuleObject::Get with same signature as MmiGet
 - ModuleObject::Set with same signature as MmiSet
+- ModuleObject::Free with same signature as MmiFree
 - Public constructor and destructor.
 
-Each ModuleObject instance knows about its client session only.
+The full internal implementation of the MMI calls is into the static library, including input argument validation, logging, etc. The reason for this is to maximize test coverage as the unit-tests for the module links to this same static library as the module SO.
```
Comment (AhmedBM) [inline docs/modules.md:982]: is "in" the static library

File: docs/modules.md [hunk 2]
```diff
@@ -971,37 +971,79 @@ The code for a module can be split into a static library and a shared object (SO
...
 
 ## 12.2. Module Shared Object (SO)
 
 The SO component of the module implements the MMI functions, with C signatures and C|C++ implementations.
 
-Following functions are global for all sessions:
+Each MMI function implementation is an empty shell that directly call into its respective ModuleObject method counterpart to execute.
```
Comment (AhmedBM) [inline docs/modules.md:988]: suggest reworking, not sure whats being conveyed here

File: docs/modules.md [hunk 3]
```diff
@@ -971,37 +971,79 @@ The code for a module can be split into a static library and a shared object (SO
...
+      "Payload": "{\"CommandId\":\"1\",\"ResultCode\":0,\"TextResult\":\"a.foo b.foo\",\"CurrentState\": 2}",
+      "ExpectedResult": 0,
+      "WaitSeconds": 0
+    }
+  ]
+}
+```
 
-<img src="assets/moduleutility.png" alt="OSConfig Module Utility" width=50%/>
+The Module Test Recipe is feed into the OSConfig's ModulesTest utility to be executed. For more information about ModulesTest and how to run it see [src/modules/test/README.md](../src/modules/test/README.md).
```
Comment (AhmedBM) [inline docs/modules.md:1046]: > feed

fed

File: docs/modules.md (hunk 1 above)
Comment (MariusNi) [reply docs/modules.md:982]: It's "links to". I do not see the issue

Comment (MariusNi) [reply docs/modules.md]: Reworded

**Review by AhmedBM — APPROVED** [review]
👌

**Pull request merged**